packages/odata-service-inquirer/test/unit/prompts/cap-project/cap-helper.test.ts

```diff
@@ -0,0 +1,337 @@
+                  "app": "app/",
+                  "db": "db/",
+                  "folderName": "flight",
+                  "path": "/test/mock/flight",
+                  "srv": "srv/",
+                },
+              },
+              {
+                "name": "Manually select CAP project folder path",
+                "value": "enterCapPath",
+              },
+            ]
+        `);
+        expect(findCapProjectsSpy).toHaveBeenCalledWith({ 'wsFolders': ['/test/mock/'] });
+    });
+
+    test('getCapEdmx', async () => {
+        const readCapServiceMetadataEdmxSpy = jest
+            .spyOn(sapuxProjectAccess, 'readCapServiceMetadataEdmx')
+            .mockResolvedValue(mockEdmx);
+        // Valid CAP service
+        expect(
+            await getCapEdmx({
+                projectPath: '/some/cap/project',
+                urlPath: '../some/service',
+                serviceName: 'Service'
+            })
+        ).toMatchInlineSnapshot(`"<?xml version="1.0" encoding="utf-8"?><edmx:Edmx Version="2"/>"`);
+
+        const errorHandlerSpy = jest.spyOn(errorHandler, 'logErrorMsgs');
+        const logErrorSpy = jest.spyOn(LoggerHelper.logger, 'error');
+
+        // Ensure we get errors returned from readCapServiceMetadataEdmxSpy and log them
+        readCapServiceMetadataEdmxSpy.mockRejectedValue('Cannot read metadata');
+        expect(
+            await getCapEdmx({
+                projectPath: '/test/mock/bookshop',
+                urlPath: '../srv/cat-service',
+                serviceName: 'CatalogService'
+            })
+        ).toBe(undefined);
+        // Error handler should be called with custom error message
+        expect(errorHandlerSpy).toHaveBeenCalledWith(
+            t('errors.cannotReadCapServiceMetadata', { serviceName: 'CatalogService' })
+        );
+        // Logger should be called with original error
+        expect(logErrorSpy).toHaveBeenCalledWith('Cannot read metadata');
+
+        // Check the real error returned by CDS
+        errorHandlerSpy.mockClear();
+        logErrorSpy.mockClear();
+        readCapServiceMetadataEdmxSpy.mockRestore();
+        expect(
+            await getCapEdmx({
+                projectPath: join(__dirname, 'fixtures/bookshop'),
+                urlPath: 'odata/v4/incident/',
+                serviceName: 'CatalogService'
+            })
+        ).toMatchInlineSnapshot(`undefined`);
+        // Error handler should be called with custom error message
+        expect(errorHandlerSpy).toHaveBeenCalledWith(
+            t('errors.cannotReadCapServiceMetadata', { serviceName: 'CatalogService' })
+        );
+        // Logger should be called with original error, this is unreliable as the underlying string value may be changed by CDS
+        expect(logErrorSpy).toHaveBeenCalledWith(
+            expect.objectContaining({
+                message: expect.stringContaining(" Module '@sap/cds' not installed in project")
+            } as Error)
+        );
```

Comment:
This is failing locally as `@sap/cds-dk` is installed globally. Could the global cds loading be mocked for the test?
`Error: Error while reading CAP service metadata. Path: '/SAPDevelop/open-ux-tools/packages/odata-service-inquirer/test/unit/prompts/cap-project/fixtures/bookshop', service uri: 'odata/v4/incident/', error: 'Error: The @sap/cds major version (>=3.33.1) specified in your CAP project is different to the @sap/cds version you have installed globally (7.4.1). Please run 'npm install' on your CAP project to ensure that the correct CDS version is loaded.'}`

Reply:
Removed test as unnecessarily testing another modules unit and mocking `require.resolve` seems like too much effort and should be done in `project-access` `module-loader` unit tests.